dace/sdfg/validation.py

```diff
@@ -353,6 +353,9 @@ def validate_sdfg(sdfg: 'dace.sdfg.SDFG', references: Set[int] = None, **context
                             "Arrays that use a multibank access pattern must have the size of the first dimension equal"
                             f" the number of banks and have at least 2 dimensions for array {name}", sdfg, None)
 
+        # Check for interstate edges that write to scalars or arrays
+        _no_writes_to_scalars_or_arrays_on_interstate_edges(sdfg)
```

Comment:
I suggest moving this to `validate_control_flow_region`, as it more directly makes use of the hierarchical, top-down traversal / validation. Otherwise you need to recursively go over interstate edges in this method, which causes twice the number of traversals necessary.